x/masterchef/keeper/abci.go

```diff
@@ -22,7 +22,7 @@ import (
 func (k Keeper) BeginBlocker(ctx sdk.Context) error {
 	params := k.parameterKeeper.GetParams(ctx)
 	// convert balances in taker address to elys and burn them
-	if ctx.BlockHeight()%int64(params.TakerFeeCollectionInterval) == 0 && params.EnableTakerFeeSwap {
+	if params.EnableTakerFeeSwap && params.TakerFeeCollectionInterval > 0 && ctx.BlockHeight()%int64(params.TakerFeeCollectionInterval) == 0 {
```

Comment:
This case is not possible as param setting will be blocked by gov message, we have zero check there